tests/rpc-subscribe-close.spec.ts

```diff
@@ -0,0 +1,124 @@
+    const context: RequestContext = { dwn, transport: 'ws', socketConnection };
+
+    const { jsonRpcResponse } = await handleSubscriptionsClose(
+      dwnRequest,
+      context,
+    );
+
+    expect(jsonRpcResponse.error).to.exist;
+    expect(jsonRpcResponse.error.code).to.equal(JsonRpcErrorCodes.InvalidRequest);
+    expect(jsonRpcResponse.error.message).to.equal('subscribe options do not exist');
+  });
+
+  it('should return an error if close subscription throws ConnectionSubscriptionJsonRpcIdNotFound', async function () {
+    const requestId = uuidv4();
+    const id = 'some-id';
+    const dwnRequest = createJsonRpcSubscribeRequest(requestId, 'rpc.subscribe.close', {}, id);
+    const socketConnection = sinon.createStubInstance(SocketConnection);
+    socketConnection.closeSubscription.throws(new DwnServerError(
+      DwnServerErrorCode.ConnectionSubscriptionJsonRpcIdNotFound,
+      ''
+    ));
+
+    const dwn = await getTestDwn();
+    const context: RequestContext = { dwn, transport: 'ws', socketConnection };
+
+    const { jsonRpcResponse } = await handleSubscriptionsClose(
+      dwnRequest,
+      context,
+    );
+
+    expect(jsonRpcResponse.error).to.exist;
+    expect(jsonRpcResponse.error.code).to.equal(JsonRpcErrorCodes.InvalidParams);
+    expect(jsonRpcResponse.error.message).to.equal(`subscription ${id} does not exist.`);
+  });
+
+  it('should return an error if close subscription throws ConnectionSubscriptionJsonRpcIdNotFound', async function () {
+    const requestId = uuidv4();
+    const id = 'some-id';
+    const dwnRequest = createJsonRpcSubscribeRequest(requestId, 'rpc.subscribe.close', {}, id);
+    const socketConnection = sinon.createStubInstance(SocketConnection);
+    socketConnection.closeSubscription.throws(new Error('unknown error'));
+
+    const dwn = await getTestDwn();
+    const context: RequestContext = { dwn, transport: 'ws', socketConnection };
+
+    const { jsonRpcResponse } = await handleSubscriptionsClose(
+      dwnRequest,
+      context,
+    );
+
+    expect(jsonRpcResponse.error).to.exist;
+    expect(jsonRpcResponse.error.code).to.equal(JsonRpcErrorCodes.InternalError);
+    expect(jsonRpcResponse.error.message).to.equal(`unknown subscription close error for ${id}: unknown error`);
+  });
+
+  it('should return a success', async function () {
+    const requestId = uuidv4();
+    const id = 'some-id';
+    const dwnRequest = createJsonRpcSubscribeRequest(requestId, 'rpc.subscribe.close', {}, id);
+    const socketConnection = sinon.createStubInstance(SocketConnection);
+
+    const dwn = await getTestDwn();
+    const context: RequestContext = { dwn, transport: 'ws', socketConnection };
+
+    const { jsonRpcResponse } = await handleSubscriptionsClose(
+      dwnRequest,
+      context,
+    );
+    expect(jsonRpcResponse.error).to.not.exist;
+  });
+
+  it('handler should generate a recordId if one is not provided with the request', async function () {
```

Comment:
Sanity: did you mean request ID instead of `recordId`?

Reply:
Thanks! Good catch! Done